packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -1174,6 +1179,267 @@ contract OPContractsManagerDeployer is OPContractsManagerBase {
+                        IFaultDisputeGame.GameConstructorParams({
+                            gameType: GameTypes.SUPER_PERMISSIONED_CANNON,
+                            absolutePrestate: _input.opChainConfigs[0].absolutePrestate,
+                            maxGameDepth: _input.gameParameters.maxGameDepth,
+                            splitDepth: _input.gameParameters.splitDepth,
+                            clockExtension: _input.gameParameters.clockExtension,
+                            maxClockDuration: _input.gameParameters.maxClockDuration,
+                            vm: IBigStepper(getImplementations().mipsImpl),
+                            weth: newPermissionedDelayedWETHProxy,
+                            anchorStateRegistry: newAnchorStateRegistry,
+                            l2ChainId: 0
+                        }),
+                        _input.gameParameters.proposer,
+                        _input.gameParameters.challenger
+                    )
+                )
+            );
+
+            // Register the new SuperPermissionedDisputeGame.
+            newDisputeGameFactory.setImplementation(
+                GameTypes.SUPER_PERMISSIONED_CANNON, IDisputeGame(address(newSuperPDG))
+            );
+            newDisputeGameFactory.setInitBond(GameTypes.SUPER_PERMISSIONED_CANNON, _input.gameParameters.initBond);
+        }
+
+        // If the permissionless game is being used, set that up too.
+        if (_input.usePermissionlessGame) {
+            // Deploy a new DelayedWETH proxy for the permissionless game.
+            IDelayedWETH newPermissionlessDelayedWETHProxy = IDelayedWETH(
+                payable(
+                    deployProxy({
+                        _l2ChainId: block.timestamp,
+                        _proxyAdmin: _input.opChainConfigs[0].proxyAdmin,
+                        _saltMixer: reusableSaltMixer(_input.opChainConfigs[0]),
+                        _contractName: "DelayedWETH-Interop-Permissionless"
+                    })
+                )
+            );
+
+            // Initialize the new DelayedWETH proxy.
+            upgradeToAndCall(
+                _input.opChainConfigs[0].proxyAdmin,
+                address(newPermissionlessDelayedWETHProxy),
+                getImplementations().delayedWETHImpl,
+                abi.encodeCall(IDelayedWETH.initialize, (proxyAdminOwner, portals[0].superchainConfig()))
+            );
+
+            // Deploy the new SuperFaultDisputeGame.
+            ISuperFaultDisputeGame newSuperFDG = ISuperFaultDisputeGame(
+                Blueprint.deployFrom(
+                    blueprints().superPermissionlessDisputeGame1,
+                    blueprints().superPermissionlessDisputeGame2,
+                    computeSalt(block.timestamp, reusableSaltMixer(_input.opChainConfigs[0]), "SuperFaultDisputeGame"),
+                    encodePermissionlessFDGConstructor(
+                        IFaultDisputeGame.GameConstructorParams({
```

Comment:
Do you think its worth switching over to the proper interface just for type checking or anything similar in the future?